godot-core/src/builtin/string/gstring.rs

```diff
@@ -123,19 +119,26 @@ impl GString {
     }
 
     ffi_methods! {
-        type sys::GDExtensionStringPtr = *mut Opaque;
+        type sys::GDExtensionStringPtr = *mut Self;
 
-        fn from_string_sys = from_sys;
-        fn from_string_sys_init = from_sys_init;
+        fn new_from_string_sys = new_from_sys;
+        fn new_with_string_uninit = new_with_uninit;
         fn string_sys = sys;
+        fn string_sys_mut = sys_mut;
     }
 
-    /// Move `self` into a system pointer. This transfers ownership and thus does not call the destructor.
+    /// Moves this string into a string sys pointer. This is the same as using [`GodotFfi::move_return_ptr`].
     ///
     /// # Safety
-    /// `dst` must be a pointer to a `GString` which is suitable for ffi with Godot.
-    pub(crate) unsafe fn move_string_ptr(self, dst: sys::GDExtensionStringPtr) {
-        self.move_return_ptr(dst as *mut _, sys::PtrcallType::Standard);
+    ///
+    /// `dst` must be a valid string pointer.
+    pub(crate) unsafe fn move_into_string_ptr(self, dst: sys::GDExtensionStringPtr) {
+        let dst: sys::GDExtensionTypePtr = dst.cast();
+        // SAFETY: `dst` is a valid string pointer. Additionally `String` doesn't behave differently for `Standard` and `Virtual`
+        // pointer calls.
+        unsafe {
+            self.move_return_ptr(dst, sys::PtrcallType::Standard);
+        }
```

Comment:
that's fair, in my wip for the script_instance i did end up putting a `#[deny(unsafe_op_in_unsafe_fn)]` on the module that has all the ffi-callbacks so that the compiler will actually fail to compile when this isn't done. because otherwise it would just never complain about any missing unsafe docs in that particular module. 

maybe something like that can be a thing we can do? in cases where there are a lot of unsafe functions we can put that deny on either the containing module/function to help with this, at least until edition 2024 when it'll be warn-by-default. at which point it'll cause compilation to fail for us since we deny warnings.

Reply:
> in my wip for the script_instance i did end up putting a `#[deny(unsafe_op_in_unsafe_fn)]` on the module that has all the ffi-callbacks so that the compiler will actually fail to compile when this isn't done

Just today I was wondering if there's a lint that would allow us to selectively opt in to that -- very cool to see that exists! I think for complex functions that have lots of mixed safe/unsafe statements, we could start using it. But I'd be careful to not overuse it for trivial ones 🙂

Reply:
yeah, that's the specific lint that will be made warn-by-default in edition 2024